zc_common/email.py

```diff
@@ -0,0 +1,102 @@
+from datetime import date
+import time
+import uuid
+
+import boto
+from boto.s3.key import Key
+from django.conf import settings
+from raven.contrib.django.raven_compat.models import client
+from zc_common.events.emit import emit_microservice_event
+
+
+S3_BUCKET_NAME = 'zc-email-template-test'
+EMAIL_EVENT_TYPE = 'send_email'
+ATTACHMENT_PREFIX = 'attachment_'
+
+
+class MissingCredentialsError(Exception):
+    pass
+
+
+def send_error_to_sentry():
+    try:
+        client.captureException()
+    except:
+        pass
+
+
+def get_s3_email_bucket():
+    aws_access_key_id = settings.AWS_ACCESS_KEY_ID
+    aws_secret_access_key = settings.AWS_SECRET_ACCESS_KEY
+    if not (aws_access_key_id and aws_secret_access_key):
+        raise MissingCredentialsError
+
+    conn = boto.connect_s3(aws_access_key_id, aws_secret_access_key)
+    bucket = conn.get_bucket(S3_BUCKET_NAME)
+    k = Key(bucket)
+    return k
+
+
+def generate_s3_folder_name():
+    email_date = date.today().isoformat()
+    email_timestamp = int(time.time())
+    email_uuid = uuid.uuid4()
```

Comment:
This is only helpful if we can also use it to generate logs, so emails are trackable in our system